tests/phpunit/integration/Core/Storage/Data_EncryptionTest.php

```diff
@@ -53,5 +53,17 @@ public function test_decrypt() {
 		$decrypted_value = $encryption->decrypt( $encrypted_value );
 
 		$this->assertEquals( 'test-value', $decrypted_value );
+
+		// Test with non string value, it should return the original value.
+		$array_value     = array( 'key' => 'value' );
+		$decrypted_value = $encryption->decrypt( $array_value );
+
+		$this->assertEquals( $array_value, $decrypted_value );
+
+		// Non-encrypted strings should be returned unmodified, without errors.
+		$string_value    = 'test-value';
+		$decrypted_value = $encryption->decrypt( $string_value );
```

Comment:
The test reads correctly to me but would this not fail as currently defined since `base64_decode( 'test-value', true ) === false` – it would invoke `error_log` ?

Actually I see that it doesn't, it just writes to the log. The more appropriate call would be to use `trigger_error` (which will fail the test as expected) but this part will be removed anyways.

Reply:
Yes correct, I was only writing it into the error log, as a way of spotting it in development. But it is removed now. Thanks